docs/tutorials/tpu-cluster-setup.md

```diff
@@ -0,0 +1,150 @@
+# Setting up a TPU Cluster
+
+This guide will walk you through the steps to set up and manage a TPU cluster using [Google Cloud Platform (GCP)](https://cloud.google.com/tpu/docs/ray-guide) and [Ray](https://docs.ray.io/en/latest/cluster/vms/user-guides/launching-clusters/gcp.html). Currently, this is the only way to use multinode TPU (i.e. more than a v4-8) with Marin.
+## Prerequisites
+
+Before you begin, ensure you have:
+- Completed the [basic installation](installation.md)
+- Access to a Google Cloud project with TPU quota
+- [Google Cloud SDK](https://cloud.google.com/sdk/docs/install) installed
+- [Ray](https://docs.ray.io/en/latest/installation.html) installed
+
+## TPU Cluster Architecture
+
+A typical TPU cluster consists of:
+- **Head Node**: A persistent GCP VM that coordinates the cluster
+- **Worker Nodes**: Autoscaling TPU VMs that execute the actual computation
+
+## Cluster Configuration
+
+The cluster configuration is defined in a [YAML template file](../../infra/marin-cluster-template.yaml). This template allows you to customize various aspects of your cluster:
+
+- Machine types for head and worker nodes
+- Number of CPUs and amount of RAM
+- Disk size and type
+- TPU types and versions
+- Minimum and maximum worker counts
+- Docker images and initialization commands
+- Network configuration
+- Environment variables and secrets. We use [GCP secrets manager](https://cloud.google.com/sdk/gcloud/reference/secrets) to store secrets.
+
+For our clusters, each config is in a separate file in the `infra` directory. These files are automatically generated by the
+`infra/update-cluster-configs.py` script, which reads the `infra/marin-cluster-template.yaml` file.
+More information can be found in the [infra/README.md](../../infra/README.md).
+
+## Building and Uploading the Docker Image
+
+We use docker images to run the jobs inside the ray cluster. We provide a convieninet make target `make cluster_docker` to build the docker image and upload it to GCP artifact registry for various different regions. This make target performs the following steps:
```

Comment:
as written `make cluster_docker` won't work for anyone else because it only uploads to our docker repo so they will need to do more work (or we need to make another script...)